tests/p2p/test_sync.py

```diff
@@ -477,6 +479,271 @@ def test_downloader_disconnect(self):
 class SyncV2HathorSyncMethodsTestCase(unittest.SyncV2Params, BaseHathorSyncMethodsTestCase):
     __test__ = True
 
+    def test_sync_metadata(self):
+        # test if the synced peer will build all tx metadata correctly
+
+        height = 0
+        # add a mix of blocks and transactions
+        height += len(self._add_new_blocks(8))
+        height += len(add_blocks_unlock_reward(self.manager1))
+        self._add_new_transactions(2)
+        height += len(self._add_new_blocks(1))
+        self._add_new_transactions(4)
+        height += len(self._add_new_blocks(2))
+        self._add_new_transactions(2)
+
+        manager2 = self.create_peer(self.network)
+        self.assertEqual(manager2.state, manager2.NodeState.READY)
+        conn = FakeConnection(self.manager1, manager2)
+
+        for _ in range(100):
+            if conn.is_empty():
+                break
+            conn.run_one_step(debug=True)
+            self.clock.advance(1)
+
+        # check they have the same consensus
+        node_sync1 = conn.proto1.state.sync_manager
+        node_sync2 = conn.proto2.state.sync_manager
+        self.assertEqual(node_sync1.peer_height, height)
+        self.assertEqual(node_sync1.synced_height, height)
+        self.assertEqual(node_sync2.peer_height, height)
+        # 3 genesis + blocks + 8 txs
+        self.assertEqual(self.manager1.tx_storage.get_vertices_count(), height + 11)
+        self.assertEqual(manager2.tx_storage.get_vertices_count(), height + 11)
+        self.assertConsensusValid(self.manager1)
+        self.assertConsensusValid(manager2)
+        self.assertConsensusEqual(self.manager1, manager2)
+
+        # Nodes are synced. Make sure manager2 has the correct metadata.
+        for tx in self.manager1.tx_storage.topological_iterator():
```

Comment:
Move this to a `assetEqualMetadata()`? I feel we might add this verification in other tests.